confluent_kafka/schema_registry/schema_registry_client.py

```diff
@@ -0,0 +1,701 @@
+        """
+        Returns ``schema`` registration information for ``subject``.
+
+        Args:
+            subject_name (str): Subject name the schema is registered under
+            schema (Schema): Schema instance.
+
+        Returns:
+            RegisteredSchema: Subject registration information for this schema.
+
+        Raises:
+            SchemaRegistryError: If schema or subject can't be found
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#post--subjects-(string-%20subject)
+
+        """  # noqa: E501
+
+        request = {'schema': schema.schema_str}
+
+        # CP 5.5 adds new fields (for JSON and Protobuf).
+        if len(schema.references) > 0 or schema.schema_type != 'AVRO':
+            request['schemaType'] = schema.schema_type
+            request['references'] = schema.references
+
+        response = self._rest_client.post('subjects/{}'
+                                          .format(_urlencode(subject_name)),
+                                          body=request)
+
+        schema_type = response.get('schemaType', 'AVRO')
+
+        return RegisteredSchema(schema_id=response['id'],
+                                schema_type=schema_type,
+                                schema=Schema(response['schema'],
+                                              schema_type,
+                                              response.get('references', [])),
+                                subject=response['subject'],
+                                version=response['version'])
+
+    def get_subjects(self):
+        """
+        List all subjects registered with the Schema Registry
+
+        Returns:
+            [str]: Registered subject names
+
+        Raises:
+            SchemaRegistryError: if subjects can't be found
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#get--subjects
+
+        """  # noqa: E501
+        return self._rest_client.get('subjects')
+
+    def delete_subject(self, subject_name):
+        """
+        Deletes the specified subject and its associated compatibility level if
+        registered. It is recommended to use this API only when a topic needs
+        to be recycled or in development environments.
+
+        Args:
+            subject_name (str): subject name
+        Returns:
+            list(int): Versions deleted under this subject
+
+        Raises:
+            SchemaRegistryError: if the request was unsuccessful.
+
+        .. _Schema Registry API Reference:
+            https://docs.confluent.io/current/schema-registry/develop/api.html#delete--subjects-(string-%20subject)
+
+        """  # noqa: E501
+        return self._rest_client.delete('subjects/{}'
+                                        .format(_urlencode(subject_name)))
+
+    def get_version(self, subject_name, version=None):
```

Comment:
it would be better to have an additional method called `get_latest_version` and make the version parameter here required.